src/types/src/index.ts

```diff
@@ -429,13 +429,36 @@ export type Manifest = {
   /** npm puts this on published manifests */
   gypfile?: boolean
   /** the author of a package */
-  author?: Person
+  author?: Person | NormalizedContributor
   /** contributors to the package */
   contributors?: NormalizedContributor[]
   /** the license of the package */
   license?: string
 }
 
+/**
+ * Utility type that overrides specific properties of type T with new types from R.
+ * Constrains override values to exclude undefined, ensuring that normalization
+ * cannot introduce undefined to fields that shouldn't have it.
+ */
+export type Override<
+  T,
+  R extends { [K in keyof R]: R[K] extends undefined ? never : R[K] },
+> = {
+  [K in keyof T]: K extends keyof R ? R[K] : T[K]
+}
+
+export type NormalizedManifest = Override<
+  Manifest,
+  {
+    bugs: NormalizedBugsEntry[]
+    author: NormalizedContributor
+    contributors: NormalizedContributor[]
+    funding: NormalizedFundingEntry[]
+    keywords: NormalizedKeywords
+  }
+>
+
```

Comment:
I think an easier to read and simpler to maintain version of it would be to define `type NormalizedManifest` as being just picking the properties that don't have a "normalized" variation, e.g: `Pick<Manifest, "name", "version", "description", ...>`  and add the normalized values to it, e.g: 
```ts
Pick<Manifest, "..."> & {
  bugs: NormalizedBugsEntry[]
  author: NormalizedContributor
  contributors: NormalizedContributor[]
  funding: NormalizedFundingEntry[]
  keywords: NormalizedKeywords
}
```